members/views.py

```diff
@@ -20,7 +20,7 @@ def form_valid(self, form):
         username = generate_username()[0]
         exists = User.objects.filter(username=username)
         if exists:
-            username = '{0}{1}'.format(generate_username()[0], randint(1, 999))
+            username = f'{generate_username()[0]}{randint(1, 999)}'
```

Comment:
This should probably be in a loop to keep trying a few times.

I would probably not use an f-string here, either, to avoid recalculating `generate_username()` every time.

Reply:
The rationale for this code is the following:

The feature allows people to enter their email address to subscribe and become a member.
We take that email address and create a new `User` object if the email address is unique in the system.
The `User` object requires the username field to be populated, but we are not asking people for an username. I installed the library that allows us to run the `generate_username` function to populate the username field when the `User` object was created.

I realized that there was a chance that the `random_username` function would return an existing username. That's why I go and run it again and then append a random 3 digit number at the end.

This is certainly smelly and I was aware of that while writing it. Possible improvements/options are:

- put the the code

```        
username = generate_username()[0]
exists = User.objects.filter(username=username)
```

inside of a loop and have run until the username ends up being unique.

My choice at the time against the idea is a personal bias against while loops run during web requests (given that there's a small possibility that it can cause a timeout error).

- Run `generate_username` and append the random 3 digit number at the end. This greatly reduces the chances of collisions. 

- Modify the feature and let people pick their own usernames. This would remove the need for the random username generation, but would require us to do a sanity check on user provided usernames (no profanity, etc).

I left it as is knowing full well that you would ask about it. I'm not against refactoring it and wanted to get your opinion.

Reply:
If we don't care about the username and no one is ever going to enter them, what about just using a UUID? UUID type 4 should always be unique.

Reply:
Yes, good point. Only reason I did not go with UUID 4 was that they are not meant to be read.
But you make a good point. Using the built in UUID 4 module will allow the code to be simpler and remove a dependency.

Could we merge in this code and then open a new PR for the updates to use an UUID 4?

Reply:
Yeah, let's go ahead with this and deal with the usernames separately.